### ISSUE TITLE
Refactor the Maven plugin 'rewrite.activeRecipes' property

### ISSUE THREAD

**Issue by ghusta** [openrewrite/rewrite-maven-plugin#722]
## TL;DR

The type of `rewriteActiveRecipes` property can be refactored to `Set<String>`.
Also, the property `activeRecipes` in class `ConfigurableRewriteMojo` should also be deprecated or deleted in favor of `rewriteActiveRecipes`, as both properties have the same role.

## Rationale

The property, defined in `ConfigurableRewriteMojo`, can be configured in differents ways : with Maven CLI, or within Maven POM in XML.

However, in `ConfigurableRewriteMojo`, there are 2 fields to manage it : `activeRecipes` and `rewriteActiveRecipes`, which could be simplified.

When printing the plugin help, both properties are displayed, which can be confusing. No one is marked deprecated.

> mvn help:describe -Dplugin=org.openrewrite.maven:rewrite-maven-plugin:5.21.0 -Dgoal=run -Ddetail

## Recommended use

Here are the expected ways to configure it (according to https://docs.openrewrite.org/reference/rewrite-maven-plugin) :

### CLI

> mvn rewrite:run -D**rewrite.activeRecipes**=an.example.Recipe0,an.example.Recipe1

### POM

```xml
<configuration>
    <activeRecipes>
        <recipe>org.openrewrite.java.Spring</recipe>
        <recipe>org.openrewrite.maven.BestPractices</recipe>
    </activeRecipes>
</configuration>
```

## Not recommended use

However, as there are 2 fields to manage this property, this is also possible at the moment 😕.
These options are possible, but should not be recommended.

### CLI

> mvn rewrite:run -D**activeRecipes**=an.example.Recipe0,an.example.Recipe1

### POM

```xml
<configuration>
    <rewriteActiveRecipes>
        <recipe>org.openrewrite.java.Spring</recipe>
        <recipe>org.openrewrite.maven.BestPractices</recipe>
    </rewriteActiveRecipes>
</configuration>
```

## Worst situation

We could even have both properties defined, which is confusing 🤯.

### CLI

> mvn rewrite:run -D**rewrite.activeRecipes**=an.example.Recipe0 -D**activeRecipes**=an.example.Recipe1

## Maven plugin working

Maven is able to manage Java collections in mojos. It can use Collection, List, Set, etc. to map properties values.
There are some corner cases to be aware of : handling of null values / superfluous spaces.
It means that somes properties still need to be sanitized in Mojo.

The good news is that comma-separated lists can be handled by Maven and transformed to collections, using CLI or POM.

So instead of :

```java
    @Nullable
    @Parameter(property = "rewrite.activeRecipes")
    protected String activeRecipes;
```

we can use :

```java
    @Nullable
    @Parameter(property = "rewrite.activeRecipes")
    protected Set<String> activeRecipes;
```

Sources:

- https://maven.apache.org/guides/plugin/guide-java-plugin-development.html
- https://maven.apache.org/guides/mini/guide-configuring-plugins.html#mapping-collection-types
- https://issues.apache.org/jira/browse/MNG-5440 (Since Maven 3.3.9 (MNG-5440), you can list individual items alternatively as **comma-separated list** in the XML value of animals directly. This approach is also used if configuring collection/array parameters via command line)

### A few examples

Given this parameter definition :

```java
    @Parameter(property = "items")
    protected Set<String> items;
```

Maven will parse these configurations this way :

> mvn goal -Ditems=a,,b

results in : [null, "a", "b"]

> mvn goal -Ditems="a , b, c"

results in : ["a ", " b", "c"]

> mvn goal -Ditems="a,a,a"

results in : ["a"]

A XML config with `<items><it/></items>`

results in : [null]

A XML config with `<items/>`

results in : []

## TODO

- Check property name used in IT with `@SystemProperty`, like in `org.openrewrite.maven.RewriteDryRunIT#no_plugin_in_pom`

**Comment by timtebeek** [comment]
Thanks for the detailed analysis and links to sources here! really helpful to give context to any changes we'll make here.
